### PR TITLE
Use existing button styles in include projects modal and sidebar

### PR DESCRIPTION
**Pull request by b12f**
The buttons in project include were using new styles, which caused
visual inconsistencies when compared to the rest of the application.
This commit changes those `spot-button`s to old `button`s. It also makes
sure that `spot-link` uses the theming variables, at least for the
non-danger variant.

Closes https://community.openproject.org/projects/openproject/work_packages/42251/activity

### CODE REVIEW

**Comment by lindenthal** [comment]
Hi @b12f,
thanks for working on this. The PP build is failing. Can you please check if this is related to this change?
Thanks
Niels

**Comment by b12f** [comment]
@lindenthal it was related to a missing backend dependency, @cbliard was keen enough to notice and has pushed a fix. It should be available soon.

**Comment by lindenthal** [comment]
hi @b12f,

I had a brief look at at the modal on iOS


image

It looks different when compared with the user invite modal. 

image


Can you please check?



Thanks
Niels

**Comment by b12f** [comment]
Hi @lindenthal this is intended. There are currently two patterns identified when it comes to modals:

1. context-dependent dropdowns (currently called `spot-drop-modal`) which the project-include modal is an instance of [figma link here](https://www.figma.com/file/XhCsrvs6rePifqbBpKYRWD/Components-Library?node-id=1408%3A6714), and
2. whole-screen modals (currently called `op-modal`). Even though there are already some ideas about how we might refactor them in the future, they have been left untouched. `spot-overlay-modal` has [some figma designs](https://www.figma.com/file/XhCsrvs6rePifqbBpKYRWD/Components-Library?node-id=625%3A4603), but there is nothing that has changed in code.

**Comment by marcalcobe** [comment]
Hi @b12f and @lindenthal,

After discussing this in the product team we've decided to only fix the distances between the buttons in the include projects modal as they are not correct. For the close/cancel action for the moment we will keep the implementations as they are as this are two different cases (full screen modal vs dropdown) and in the near future we will propose an aligned solution for mobile in this cases.

File: frontend/src/app/spot/styles/sass/components/link.sass [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .spot-link
   @include unset-button-styles
   display: inline
-  color: $spot-color-main
+  color: var(--content-link-color)
```
Comment (HDinger) [inline frontend/src/app/spot/styles/sass/components/link.sass]: This is not enough as it brings us back to the problem we originally wanted to solve with the new links. For the default theme (as well as the light theme by the way) the `content-link-hover-active-color` is the same as `content-link-color` resulting in no visible hover effect except the underlining. So with the theme colours we need the underlining as well.

**Review by HDinger — CHANGES_REQUESTED** [review]
I have some remarks:
* You forgot to replace the sidebar button
Bildschirmfoto 2022-05-13 um 12 44 29
 
* There is an issue with the link hover effect (see my comment).
* For mobile there is also still the new button visible and I think the margins don't match the specification.
Bildschirmfoto 2022-05-13 um 12 47 18

**vs**
Bildschirmfoto 2022-05-13 um 12 49 50


* Please also have a look at the merge conflict in the Gemfile.

File: frontend/src/app/spot/styles/sass/components/link.sass (hunk 1 above)
Comment (marcalcobe) [reply frontend/src/app/spot/styles/sass/components/link.sass]: Hi @b12f and @HDinger,

Sorry for the confusion on the mobile modal again! We where doing some trials with the Product Team on how we could harmonise the buttons in all the modals in the complete application. Please don't actualise anything in regards of this full width button on the mobile version. Also there is an error also solved where the clear selection button says cancel, please keep the clear selection option.

**Review by HDinger — CHANGES_REQUESTED** [review]
I think you have accidentally overwritten parts of your older changes. For me it looks currently like this: 

Bildschirmfoto 2022-05-17 um 09 41 17

The sidebar button is correct. However, the project include again uses the spot-buttons. Further, the link to add a new assignee has the wrong color (although the hover effect is correct).

**Comment by HDinger** [comment]
One additional remark: This should be targeted agains the `release/12.1` branch in order to include it to the next patch release.

**Comment by b12f** [comment]
Really weird, I did a rebase here and that apparently removed the commit that did the initial button changes. I'll see how I can fix this.

**Comment by b12f** [comment]
Superseded by https://github.com/opf/openproject/pull/10692

**Pull request closed without merging**